File: RETOS/Reto30/Reto30.java
package RETOS.Reto30;

import java.util.Scanner;

public class Reto30 {


    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Introduce el mensaje: ");
        String mensaje = scanner.nextLine();
        System.out.print("Introduce la clave: ");
        int clave = scanner.nextInt();
        String mensajeEncriptado = encriptarMensaje(mensaje, clave);
        System.out.println("Mensaje encriptado: " + mensajeEncriptado);
    }

    public static String encriptarMensaje(String mensaje, int clave) {
        StringBuilder mensajeEncriptado = new StringBuilder();
        clave = clave % 26; 
        for (int i = 0; i < mensaje.length(); i++) {
            char caracter = mensaje.charAt(i);
            if (Character.isLetter(caracter)) {
                char base = Character.isLowerCase(caracter) ? 'a' : 'A';
                caracter = (char) (base + (caracter - base + clave) % 26);
            }
            mensajeEncriptado.append(caracter);
        }
        return mensajeEncriptado.toString();
    }
}

File: RETOS/Reto30/Reto30.kt
fun encriptarMensaje(mensaje: String, clave: Int): String {
    val mensajeEncriptado = StringBuilder()
    var claveMod = clave % 26 

    for (caracter in mensaje) {
        if (caracter.isLetter()) {
            val base = if (caracter.isLowerCase()) 'a' else 'A'
            val nuevoCaracter = (base + (caracter - base + claveMod) % 26)
            mensajeEncriptado.append(nuevoCaracter)
        } else {
            mensajeEncriptado.append(caracter)
        }
    }
    return mensajeEncriptado.toString()
}

fun main() {

    println("Introduce el mensaje: ")
    val mensaje = readLine() ?: ""
    println("Introduce la clave: ")
    val clave = readLine()?.toIntOrNull() ?: 0
    val mensajeEncriptado = encriptarMensaje(mensaje, clave)
    println("Mensaje encriptado: $mensajeEncriptado")
}
